nuxeo-notification-stream-core/src/main/java/org/nuxeo/ecm/platform/notification/notifier/Notifier.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Contributors:
+ *      Nuxeo
+ */
+
+package org.nuxeo.ecm.platform.notification.notifier;
+
+import static org.nuxeo.runtime.stream.StreamServiceImpl.DEFAULT_CODEC;
+
+import java.util.Map;
+
+import org.nuxeo.ecm.platform.notification.NotificationSettingsService;
+import org.nuxeo.ecm.platform.notification.message.Notification;
+import org.nuxeo.lib.stream.computation.AbstractComputation;
+import org.nuxeo.lib.stream.computation.ComputationContext;
+import org.nuxeo.lib.stream.computation.Record;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.codec.CodecService;
+
+/**
+ * @since XXX
+ */
+public abstract class Notifier extends AbstractComputation {
+
+    protected Map<String, String> properties;
+
+    public Notifier(NotifierDescriptor desc) {
+        super(desc.id, 1, 0);
+        properties = desc.getProperties();
+    }
+
+    @Override
+    public void processRecord(ComputationContext ctx, String inputStreamName, Record record) {
+
+        Notification notification = Framework.getService(CodecService.class)
```

Comment:
you can use a codec field initialized during computation init, so you don't have to do this on each record

Reply:
:+1: